src/dag-service.js

```diff
@@ -23,10 +25,39 @@ function DAGService (blockService) {
   // this.addRecursive
 
   // get retrieves a DAGNode, using the Block Service
-  this.get = (multihash, callback) => {
-    if (!multihash) { return callback(new Error('Invalid Key')) }
+  this.get = function (multihash, callback) {
+    const isBuf = Buffer.isBuffer(multihash)
+    const isString = typeof multihash === 'string'
+
+    if (!isBuf && !isString) {
+      return callback(new Error('Invalid Key'))
+    }
+
+    if (isBuf) {
+      var mhString = base58.encode(multihash)
+      if (!isIPFS.multihash(mhString)) { return callback(new Error('Invalid Key')) }
+      this.getWith(multihash, callback)
+    }
+
+    if (isString) {
+      var isMhash = isIPFS.multihash(multihash)
+      var isPath = isIPFS.path(multihash)
+      if (!isMhash && !isPath) {
+        return callback(new Error('Invalid Key'))
+      }
+      if (isMhash) {
+        var mhBuffer = new Buffer(base58.decode(multihash))
+        this.getWith(mhBuffer, callback)
+      }
+      if (isPath) {
+        var ipfsKey = new Buffer(base58.decode(multihash.replace('/ipfs/', '')))
+        this.getWith(ipfsKey, callback)
+      }
+    }
+  }
```

Comment:
:+1: @nginnever :D 

Can we get all of this part of 'is-ipfs' logic? I'm sure @xicombd would be happy to include it there and it will be useful for a ton of other places.

btw, it would be stellar if [mafmt](https://www.npmjs.com/package/mafmt) from @whyrusleeping used this to check multiaddrs that are also ipfs multiaddrs with valid multihashes :)